lib/thredded.rb

```diff
@@ -143,7 +143,7 @@ def self.user_class=(user_class_name)
     unless user_class_name.is_a?(String)
       fail "Thredded.user_class must be set to a String, got #{user_class_name.inspect}"
     end
-    @@user_class_name = user_class_name # rubocop:disable Style/ClassVars
+    @@user_class_name = user_class_name.demodulize # rubocop:disable Style/ClassVars
```

Comment:
This is incorrect as it may no longer refers to the same class. E.g. `"Spree::User".demodulize` is "User" (which could be `::User` etc).